api/src/test/java/org/apache/iceberg/types/TestTypes.java

```diff
@@ -43,6 +67,13 @@ public void fromPrimitiveString() {
 
     assertThat(Types.fromPrimitiveString("Decimal(2,3)")).isEqualTo(Types.DecimalType.of(2, 3));
 
+    assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(() -> Types.fromPrimitiveString("variant"))
+        .withMessage("Cannot parse type string to primitive: variant");
```

Comment:
I think this message could be confusing. When this happens, we know that the type name is valid. The problem is that it can't be returned as a `PrimitiveType`. I would prefer to say that in the error: `"Cannot parse type string: variant is not a primitive type"`.